pkg/auth/cred.go

```diff
@@ -17,19 +17,42 @@ limitations under the License.
 package auth
 
 import (
+	"context"
 	"fmt"
+	"time"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"k8s.io/klog/v2"
+	"knative.dev/pkg/logging"
 )
 
+type TokenWrapper struct {
+	cred azcore.TokenCredential
+}
+
+func NewTokenWrapper(cred azcore.TokenCredential) *TokenWrapper {
+	return &TokenWrapper{
+		cred: cred,
+	}
+}
+
+func (w *TokenWrapper) GetToken(ctx context.Context, options policy.TokenRequestOptions) (azcore.AccessToken, error) {
+	token, err := w.cred.GetToken(ctx, options)
+	if err != nil {
+		return azcore.AccessToken{}, err
+	}
+	logging.FromContext(ctx).Info("refreshing MDAL Token")
+	token.ExpiresOn = time.Now().Add(2 * time.Hour)
```

Comment:
Do we need to check the original `ExpiresOn` in case it’s already less than 2 hours?

Reply:
I think that would be prudent

Reply:
IIRC the underlying token credential attempts to [fetch from the cache first](https://github.com/Azure/azure-sdk-for-go/blob/93e07a3a4ef307a418a8090595aa2fb88d227083/sdk/azidentity/confidential_client.go#L104). It seems like it could return a short-lived token?

Reply:
yep this value should be set with the minimum value of token expires on and the 2hr we want to use here.